x-pack/plugins/security/server/authentication/unauthenticated_page.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// @ts-expect-error no definitions in component folder
+import { EuiButton } from '@elastic/eui/lib/components/button';
+import React from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import type { IBasePath } from 'src/core/server';
+
+import { PromptPage } from '../prompt_page';
+
+interface Props {
+  originalURL: string;
+  buildNumber: number;
+  basePath: IBasePath;
+}
+
+export function UnauthenticatedPage({ basePath, originalURL, buildNumber }: Props) {
+  return (
+    <PromptPage
+      buildNumber={buildNumber}
+      basePath={basePath}
+      title={i18n.translate('xpack.security.unauthorized.pageTitle', {
```

Comment:
__question:__ Call for suggestions

So here is the most complex part - the text.

* Login page/selector - here we just have a single line of the text. I'm not sure if we can make it super useful or actionable (e.g. add something like  `contact administrator` or `check the logs`), so I just kept it short for now. 

  My main concern is that the current wording may work well for the case when we get 401 during initial login, but it will be confusing in any other case, maybe something like `We couldn't authenticate you with the credentials provided. Please try again.` (borrowed from the MS products error messages) will be better?

![Screen Shot 2021-04-22 at 08 57 08](https://user-images.githubusercontent.com/1713708/115510134-a7852800-a27f-11eb-82ad-2959b3800c20.png)

* Dedicated unauthenticated page (when login selector/page isn't available) - the body text should most likely be the same as we'll use for the login page, but I don't have much ideas for title - I've seen many cases when it's just `Unauthorized`, but I agree with you that's kind of a misnomer.

![Screen Shot 2021-04-22 at 08 55 19](https://user-images.githubusercontent.com/1713708/115510149-ab18af00-a27f-11eb-846e-99f18a2478e7.png)

What do you all think?

Reply:
> **question:** Call for suggestions
> 
> So here is the most complex part - the text.

Other suggestions that come to mind:

* An exception occurred during log in, please check your credentials and try again.
* Log in failed, this can happen when single sign-on is not configured properly. Please check your credentials and try again.

I imagine that the vast majority of people who run into this will be attempting SAML or OIDC authentication, do you agree? If so, I lean towards the second message. Also perhaps @elastic/kibana-docs may want to weigh in.

Reply:
I like Microsoft's suggestion that Oleg linked above:

> We couldn't authenticate you with the credentials provided. Please try again.


@jportner,

> Other suggestions that come to mind:
> 
> * An exception occurred during log in, please check your credentials and try again.
> * Log in failed, this can happen when single sign-on is not configured properly. Please check your credentials and try again.
> 
> I imagine that the vast majority of people who run into this will be attempting SAML or OIDC authentication, do you agree? If so, I lean towards the second message.


The scenarios we are currently catering for are SSO-related, but the implementation is handling all scenarios where a `401` is produced, either intentionally or otherwise. I wouldn't want to suggest that it was an SSO problem and send our users on a wild goose chase if it turns out to be another issue altogether.

Thinking about the Cloud experience: Whatever message we decide on here will be shown to users who attempt to login to someone else's cluster using Cloud SSO. This isn't a misconfigured SSO setup, this is simply user error. I wouldn't want our users to think that we somehow misconfigured their SSO.

Reply:
Let me rephrase--

I think "We couldn't authenticate you with the credentials provided. Please try again." could be problematic if we give no indication as to _why_ we couldn't authenticate the user, and we are essentially telling them to try the same thing again. How can we best convey to them to try something different?

Reply:
> could be problematic if we give no indication as to why we couldn't authenticate the user, and we are essentially telling them to try the same thing again. How can we best convey to them to try something different?

Good question. I'm not sure what options we have, since we don't know why we can't authenticate the user. We could provide a troubleshooting guide, but I don't love solving these types of ux problems with docs alone. I'll give this some more thought

Reply:
What if the message give more than one thing to try?  Something like:

**We couldn't log you in**

A few things to try:

- Verify your credentials
- Test your connection
- Make sure you're not blocked by security software
- Contact your system administrator

Try log in again

Reply:
Sorry I know it's confusing, but this one a is a bit tricky, let me try to give more details:

> Verify your credentials

* While it may *technically* be correct, users that provide wrong credentials at the login selector/form shouldn't see these new screens, and will end up with something like this instead:

![Screen Shot 2021-04-24 at 11 47 33](https://user-images.githubusercontent.com/1713708/115853740-c6272280-a429-11eb-8da4-7d6d2d369e95.png)

* If they enter wrong credentials at the SSO provider (e.g. Okta) they won't be even redirected to Kibana.

* If we simplify, during initial login users can see these new screens only if:
  * PKI authentication is used and browser doesn't provide a valid certificate
  * Kerberos authentication is used and browser doesn't provide a valid SPNEGO token
  * Authenticating reverse proxy sits in front of Kibana and provides wrong credentials
  * User uses some browser plugin that automatically provides wrong credentials (basically same cases as with reverse proxy)

If we put tech savvy users aside, it feels like most of the users won't be able to figure out what credentials they need to provide, and will need to contact administrator anyway.

> Test your connection

We shouldn't show these new screens if there is a problem with the connection (or more precisely the problem with the connection that user can actually fix).

> Contact your system administrator
> Try log in again

These two options sound like the most useful to me - either administrators has fixed something in the meantime already and another attempt to log in will succeed, or you need to let them know that you're having problems...

Reply:
### Login message

Responding to what @azasypkin said, how about any of these options

**Uses "log in" in the title**

> We couldn't log you in
> Please check your credentials and try again. If you still can't log in, contact your system administrator.
> Log in

**Uses "authentication" in the title**

> We hit an error authenticating your account
> Please check your credentials and try again. If you still can't log in, contact your system administrator.
> Log in

**Uses both terms**

>We couldn't log you in
>We hit an authentication error.  Please check your credentials and try again. If you still can't log in, contact your system administrator.
> Log in

### Message on login screen

Just a few nits:

Although apps have traditionally used "Invalid username or password", some are now using "Username or password is incorrect" which sounds a little friendlier to me.

See more login options > More login options

Reply:
Thanks for the suggestions @gchaps! I like the `Uses both terms` suggestion the best, but I would be happy with any of them if someone else feels strongly about one in particular.

> Although apps have traditionally used "Invalid username or password", some are now using "Username or password is incorrect" which sounds a little friendlier to me.

++ I like this change too

Reply:
`Uses both terms` sounds good to me as well, let's go with it, thanks!